libbeat/template/config.go

```diff
@@ -76,6 +77,25 @@ func DefaultConfig() TemplateConfig {
 	}
 }
 
+func Unpack(c *common.Config) (TemplateConfig, error) {
+	if c == nil {
+		return DefaultConfig(), nil
+	}
+
+	jsonEnabled, _ := c.Bool("json.enabled", -1)
+	config := DefaultConfig()
+	if jsonEnabled {
+		cfgwarn.Deprecate("8.0.0", "Please migrate your JSON templates from legacy template format to composable index template.")
+		config.Type = IndexTemplateLegacy
```

Comment:
What if the user wants to use the new templates in 7.16? Will line 93 overwrite it?

Reply:
No, it is not overwritten. The configuration from the user is unpacked later.

Reply:
That is what I mean by 93, this is where the user config is unpacked. Looks like we are on the same page.